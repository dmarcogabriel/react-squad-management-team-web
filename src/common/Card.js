import React from 'react';
import cn from 'classnames';

const Card = ({
  title,
  RightComponent,
  children,
  className = '',
  childrenClassName = '',
}) => (
  <div className={cn('bg-white rounded-lg shadow-lg my-10 w-full', className)}>
    <div className="flex p-8 justify-between items-center border-b-2 border-gray-200">
      <h1 className="text-secondary-dark font-bold text-md">{title}</h1>
      {RightComponent && <RightComponent />}
    </div>
    <div className={cn('py-5 px-2', childrenClassName)}>{children}</div>
  </div>
);

export default Card;
