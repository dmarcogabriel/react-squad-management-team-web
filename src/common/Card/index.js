import React from 'react';
import cn from 'classnames';

const Card = ({
  title,
  RightComponent,
  children,
  className = '',
  childrenClassName = '',
}) => (
  <div
    data-testid="card"
    className={cn('bg-white rounded-lg shadow-lg my-10 w-full', className)}
  >
    <div className="flex p-8 justify-between items-center border-b-2 border-gray-200">
      <h1
        data-testid="card-title"
        className="text-secondary-dark font-bold text-md"
      >
        {title}
      </h1>
      {RightComponent && <RightComponent />}
    </div>
    <div
      data-testid="card-children"
      className={cn('py-5 px-2', childrenClassName)}
    >
      {children}
    </div>
  </div>
);

export default Card;
