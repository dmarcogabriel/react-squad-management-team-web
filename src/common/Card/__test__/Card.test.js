import { render, cleanup } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Card from '..';

afterEach(cleanup);

it('renders Card without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card />, div);
});

it('renders Card classes correctly', () => {
  const { getByTestId } = render(<Card className="my-5" />);

  expect(getByTestId('card')).toHaveClass(
    'bg-white rounded-lg shadow-lg my-10 w-full my-5'
  );
});

it('renders Card title correctly', () => {
  const { getByTestId } = render(<Card title="This is the title" />);

  expect(getByTestId('card-title')).toHaveTextContent('This is the title');
});

it('renders Card children correctly', () => {
  const { getByTestId } = render(
    <Card>
      <h2>This is a children</h2>
    </Card>
  );

  expect(getByTestId('card-children')).toContainHTML(
    '<h2>This is a children</h2>'
  );
});

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <Card
        title="This is a test"
        className="my-5"
        RightComponent={() => <span>Right component</span>}
        childrenClassName="my-5"
      >
        Children test
      </Card>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
