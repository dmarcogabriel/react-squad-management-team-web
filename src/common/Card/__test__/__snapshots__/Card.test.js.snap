// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `
<div
  className="bg-white rounded-lg shadow-lg my-10 w-full my-5"
  data-testid="card"
>
  <div
    className="flex justify-between items-center p-8 border-b-2 border-gray-200"
  >
    <h1
      className="text-secondary-dark font-bold text-md"
      data-testid="card-title"
    >
      This is a test
    </h1>
    <span>
      Right component
    </span>
  </div>
  <div
    className="py-5 px-2 my-5"
    data-testid="card-children"
  >
    Children test
  </div>
</div>
`;
